keepers/keeper-core/src/lib.rs

```diff
@@ -31,15 +34,15 @@ pub type Error = Box<dyn std::error::Error>;
 #[derive(ThisError, Debug, Clone)]
 pub enum TransactionExecutionError {
     #[error("Transactions failed to execute after multiple retries")]
-    RetryError(Vec<Instruction>),
-    #[error("Transactions failed to execute after multiple retries")]
-    TransactionRetryError(Vec<Vec<Instruction>>),
+    TransactionRetryError(Vec<(Vec<Instruction>, String)>),
```

Comment:
there's other transaction types baked into the ClientError like IoError, ReqwestError, SigningError, etc that I feel is useful to propogate. and the only way to bubble it up is to cast to string since the ClientError type doesn't implement the Clone trait we need. If we saved only the transaction errors you'd end up with `Vec<Result<(), Option<TransactionError>>>`. Can switch that out if it's preferred but we'd be losing information with that